File: rb/features.rb
# VisitorFrequenciesLucerneCh SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module VisitorFrequenciesLucerneChFeatures
  def self.make_feature(name)
    case name
    when "base"
      VisitorFrequenciesLucerneChBaseFeature.new
    when "test"
      VisitorFrequenciesLucerneChTestFeature.new
    else
      VisitorFrequenciesLucerneChBaseFeature.new
    end
  end
end

File: py/visitorfrequencieslucernech_sdk/features.py
# VisitorFrequenciesLucerneCh SDK feature factory

from visitorfrequencieslucernech_sdk.feature.base_feature import VisitorFrequenciesLucerneChBaseFeature
from visitorfrequencieslucernech_sdk.feature.test_feature import VisitorFrequenciesLucerneChTestFeature


def _make_feature(name):
    features = {
        "base": lambda: VisitorFrequenciesLucerneChBaseFeature(),
        "test": lambda: VisitorFrequenciesLucerneChTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
